src/lambda/vue/samInvoke.ts

```diff
@@ -0,0 +1,501 @@
+ * Call back into the webview with the selected launch config.
+ * @param postMessageFn
+ */
+async function loadSamLaunchConfig(postMessageFn: (response: SamInvokerResponse) => Thenable<boolean>): Promise<void> {
+    // TODO: Find a better way to infer this. Might need another arg from the frontend (depends on the context in which the launch config is made?)
+    const workspaceFolder = vscode.workspace.workspaceFolders?.length ? vscode.workspace.workspaceFolders[0] : undefined
+    if (!workspaceFolder) {
+        // TODO Localize
+        vscode.window.showErrorMessage(localize('AWS.lambda.form.noFolder', 'No workspace folder found.'))
+        return
+    }
+    const uri = workspaceFolder.uri
+    const launchConfig = new LaunchConfiguration(uri)
+    const pickerItems = getLaunchConfigQuickPickItems(launchConfig, uri)
+    if (pickerItems.length === 0) {
+        pickerItems.push({
+            index: -1,
+            label: localize('AWS.lambda.form.noDebugConfigs', 'No aws-sam debug configurations found.'),
+            detail: localize('AWS.picker.dynamic.noItemsFound.detail', 'Click here to go back'),
+        })
+    }
+    const qp = picker.createQuickPick({
+        items: pickerItems,
+        options: {
+            title: localize('AWS.lambda.form.selectDebugConfig', 'Select Debug Configuration'),
+        },
+    })
+
+    const choices = await picker.promptUser({
+        picker: qp,
+    })
+    const pickerResponse = picker.verifySinglePickerOutput<LaunchConfigPickItem>(choices)
+
+    if (!pickerResponse || pickerResponse.index === -1) {
+        return
+    }
+    postMessageFn({
+        command: 'loadSamLaunchConfig',
+        data: {
+            launchConfig: pickerResponse.config!,
+        },
+    })
+}
+
+interface SampleRequestManifest {
+    requests: {
+        request: {
+            name?: string
+            filename?: string
+        }[]
+    }
+}
+
+interface SampleQuickPickItem extends vscode.QuickPickItem {
+    filename: string
+}
+
+/**
+ * Open a quick pick containing upstream sample payloads.
+ * Call back into the webview with the contents of the payload to add to the JSON field.
+ * @param postMessageFn
+ */
+async function getSamplePayload(postMessageFn: (response: SamInvokerResponse) => Thenable<boolean>): Promise<void> {
+    // stolen from invokeLambda.ts
+    try {
+        const sampleInput = await makeSampleRequestManifestResourceFetcher().get()
+
+        if (!sampleInput) {
+            throw new Error('Unable to retrieve Sample Request manifest')
+        }
+
+        getLogger().debug(`Loaded: ${sampleInput}`)
+
+        const inputs: SampleQuickPickItem[] = []
+
+        await new Promise<void>((resolve, reject) => {
+            xml2js.parseString(sampleInput, { explicitArray: false }, (err: Error, result: SampleRequestManifest) => {
+                if (err) {
+                    reject()
+                }
+
+                _.forEach(result.requests.request, r => {
```

Comment:
lodash is unnecessary here

Reply:
Wasn't a conscious use (copy/paste)...I'll see if I can remove it.